columnflow/production/cms/pdf.py

```diff
@@ -74,4 +74,24 @@ def pdf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
     events = set_ak_column_f32(events, "pdf_weight_up", 1 + stddev)
     events = set_ak_column_f32(events, "pdf_weight_down", 1 - stddev)
 
+    if ak.any(stddev > 0.5):
+        # catch events with large pdf variations
+        # TODO: how should we handle large PDF weight uncertainties (especially above 100%)?
+        logger.warning(
+            f"In dataset {self.dataset_inst.name}, there are {ak.sum(stddev > 0.5)} " +
+            "entries with pdf uncertainty above 50%",
+        )
```

Comment:
I like this summary very much, but it might be worthwhile to know the fraction of occurances w.r.t. total amount of events, like

 ```suggestion
        occurances = ak.sum(stddev>0.5)
        frac = occurances/ak.count(stddev, axis=0) * 100
        logger.warning(
            f"In dataset {self.dataset_inst.name}, there are {occurances} ({frac:.2f}%) " +
            "entries with pdf uncertainty above 50%",
        )
```

Reply:
Regarding what we should do in case the std dev is very large or the nominal weight is 0: As far as I'm aware, there isn't one global way to go about this, because it shouldn't happen in theory. Since the producer is part of the cf backend, we could think about exposing options about what to do to the user, e.g.

- throw a fatal error and displace the sample (or even file?) and the number of occurances
- just display the warning
- discard the events that are problematic

This is obviously up for debate.